### PR TITLE
FixFor#97

### LINKED ISSUE #97: Cross entropy loss produces error
The "cross-entropy" loss produces an error when trying to fit a multi-class dnn.

**Reprex:**
```
library(cito)

nn.fit <- dnn(Species ~ Sepal.Width + Sepal.Length + Petal.Length + Petal.Width, 
              data = datasets::iris,
              loss = "cross-entropy")
```
```
Error in torch_tensor_cpp(data, dtype, device, requires_grad, pin_memory) : 
  R type not handled
```

### PR DESCRIPTION
**Pull request by florianhartig**
Fixes #97

Main fix is L 33 -> add cross-entropy

I also erased L 48 because afaiks, the condition was already checked in L33 so this did seem redundant - can someone check that this is actually true?

### CODE REVIEW

**Comment by MaximilianPi** [comment]
We should remove loss = 'softmax' because it is confusing and just a alias for "cross-entropy." However, it should first be flagged as deprecated in the next release.

**Review by ArminSchenk — APPROVED** [review]
Makes sense. The redundant if clause was also already removed in my pending pull request. I will merge the changes, even though the format_targets function will soon be replaced anyways.
